Mono.Debugger.Soft/Mono.Debugger.Soft/VirtualMachineManager.cs

```diff
@@ -111,20 +111,19 @@ public static IAsyncResult BeginLaunch (ProcessStartInfo info, AsyncCallback cal
 				socket.Close ();
 			};
 
-			LaunchCallback c = new LaunchCallback (LaunchInternal);
-			return c.BeginInvoke (p, info, socket, callback, socket);
+			Task<VirtualMachine> t3 = Task<VirtualMachine>.Run (() => {
+				return LaunchInternal (p, info, socket);
+			});
+			t3.ContinueWith ((antecendent => callback (antecendent)));
+			return t3;
 		}
 
-		public static VirtualMachine EndLaunch (IAsyncResult asyncResult) {
+		public static VirtualMachine EndLaunch (Task<VirtualMachine> asyncResult) {
 			if (asyncResult == null)
 				throw new ArgumentNullException ("asyncResult");
 
-			if (!asyncResult.IsCompleted)
-				asyncResult.AsyncWaitHandle.WaitOne ();
-
-			AsyncResult result = (AsyncResult) asyncResult;
-			LaunchCallback cb = (LaunchCallback) result.AsyncDelegate;
-			return cb.EndInvoke (asyncResult);
+			asyncResult.Wait ();
```

Comment:
`.Wait()` is not needed if we're doing `.Result`.

One other remark is that we probably don't want to force a `.Result` here. Maybe we need a `public static Task<VirtualMachine> Launch()`?